tools/acorn-optimizer.mjs

```diff
@@ -1949,33 +1949,37 @@ function reattachComments(ast, comments) {
 
   // Walk through all comments in ascending line number, and match each
   // comment to the appropriate code block.
-  for (let i = 0, j = 0; i < comments.length; ++i) {
-    while (j < symbols.length && symbols[j].start.pos < comments[i].end) {
+  let j = 0;
+  for (const [pos, comments] of Object.entries(commentsMap)) {
+    while (j < symbols.length && symbols[j].start.pos < pos) {
       ++j;
     }
     if (j >= symbols.length) {
-      trace(`dropping comment: no symbol comes after it (${comments[i].value.slice(0, 30)})`);
+      trace('dropping comments: no symbol comes after them');
       break;
     }
-    if (symbols[j].start.pos - comments[i].end > 20) {
+    if (symbols[j].start.pos - pos > 20) {
```

Comment:
This was `comments[i].end`, but now as the key in that map it was set as `token.start` on line 2048. Why is that ok?

Reply:
Yes this is kind of the point of this PR.   We associate each comment not with its own end position, but instead with the start position of the next token.

This means if you have 100 comments in a row they will be associated with the same `pos` here.

Reply:
Thanks, now I see... I was thinking that part was NFC, which confused me :smile: